### ISSUE TITLE
[Feature] Tagging Flow/Interface to facilitate content tagging

### ISSUE THREAD

**Issue by quirkyjoe** [stashapp/stash#3364]
**Is your feature request related to a problem? Please describe.**
Tagging can be arduous, and it's not always obvious what media is missing important tags.

**Describe the solution you'd like**
Stash could provide a tagging interface/flow, where scenes with low tag counts are show one after another, with a tag cloud to the side from which the user can select applicable tags (and the tag entry field).
The idea would be to obviate the need to dig through the library for poorly tagged content, and instead make a fun process by which the user can watch and add tags.

The tag cloud could be populated with common tags, along with a random assortment.
(I'm not talking any sort of visual recognition, just simple/random suggestions and the usual tag entry field.)
Ex:
"Does this scene contain:
Brunette / Solo / Toys / Thicc / Etc."

**Additional context**
It could look something like the moderation flow on bunnyfap.com:
https://bunnyfap.com/moderation/moderation
![image](https://user-images.githubusercontent.com/85048415/213084504-b0ee550d-42fc-41ce-b672-7d6d72fa810e.png)

**Comment by LeGrosFromage** [comment]
This does sound extremely useful.  I hate tagging, I never apply more than 2 or 3 tags when I manually edit a scene (that's even if I do any at all).  I think they're overrated.  That being said, with a more visual approach as shown above, then yes it would be easy to add 5-10 relevant tags or so rather than trying to remember what tags you already have or ones that are similar to what you think they should be.  It's a solid idea.

**Comment by quirkyjoe** [comment]
I think tag categories (and other tag improvements) are needed and valuable, and indeed would make tagging easier.

I do think that those improvements are orthogonal to my suggestion though.

Having an engaging UI to motivate tagging (and surface content that needs tagging) would simply compliment those improvements. I hope this would make tagging a fun activity, rather than a chore (even if that chore is made easier).

I found this to be the case for the moderation UI I mentioned in my initial post.

**Comment by ALonelyJuicebox** [comment]
Right, but I have all of what you've mentioned _today_ using (pseudo) tag categories. Positions, orgasms and where they happen, all of it sorted in an appropriate category.
 I'm all for UX improvements, but I'm struggling a bit with your vision here...🤔

**Comment by Stashing-fr** [comment]
Yes I do it differently too, we adapt ^^

My message was about how to generate the GUI if this solution was chosen.
By allowing users to easily reuse the structure that links the tags in the database it would create a GUI interface that adapts to each database.
The advantage being to quickly be able to configure an interface which contains all the child nodes by indicating only the desired parent nodes.
It would also allow the interface to update automatically based on changes in tag relationships without having to reconfigure the GUI manually.

The GUI allows a form of exhaustiveness to avoid oversights, by browsing quickly in an organized way through the tags. Adding tags is also much faster this way.
This is something difficult and much longer to do with the current system. And the current modes of organization are mostly limited to hierarchical classifications.

It was just an idea.
I also agree with what has been said, on the fact that it would be useful to have an interface more suited to tagging, by reducing the size of the video and increasing the space for tagging.


I would like to take the time to develop several proposals about tags, which sometimes allow to connect several ideas already evoked by others in a coherent whole. I will think about this and share a clear and illustrated presentation in order to get feedback.

**Comment by 0xb0af** [comment]
I agree with @quirkyjoe that improving the UX for manual tagging is orthogonal to how tags are structured/nested.

I would suggest thinking about this in terms of "In the current UX, how many clicks and mouse moves and/or keystrokes are required to go from a single untagged video to adding a large number of tags (5-10)."

As an example, I created a search filter that shows all untagged scenes in random order. 

I click on the first untagged scene to open it. I move the mouse to the Edit tab and click. The sidebar is not tall enough to show all the way down to Tags so I move the mouse to the sidebar and scroll it down, then click the empty box under Select Tags. I start typing the tags I'm interested in one by one (the find-as-you-type is super nice here). After each one I press Return to add it to the list and start typing the next. If a tag doesn't exist, I can create it, which is also nice. Once I've added all the tags, I scroll back up to make the Save button visible, then I move the mouse and click it. 

Then I go back to the scene list and start over on the next video.  This is way too much effort when you multiply it by dozens or hundreds of scenes.

For someone who tags many scenes by hand, every keystroke/mouse interaction that can be eliminated here  is a big win.

My ideal flow would include:
- A 'tagging' mode that simply shows one untagged video after another. A single keypress/click to indicate "I'm done tagging this video, drop me directly into tagging the next"
- The ability to easily add tags and have them saved immediately. I don't even think a Save button is necessary for an operation that's as trivial to undo as adding a tag.
- Entirely keyboard driven. Type a tag to enter it. Tab/return to autocomplete and save it. Escape to abort this tag. Shift+Return to say "All tags done, next video please"

Some nice-to-haves would be:
- Automatic suggestion of tags, either based on commonly used, previously used, or often-used-together
- A way to tag multiple videos with the same group of tags.
- The ability to customize the filter used to determine what is considered an 'untagged' video (ie. maybe I want anything with less than 3 tags to show up in the flow).

These are just suggestions.  I'm less concerned with the final look/layout and more with how it functions. Any sort of dedicated workflow that allows me to add many tags easily with minimum interaction will be a winner in my book.
